core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataCutter.scala

```diff
@@ -121,25 +166,30 @@ class DataCutter(uid: String = UID[DataCutter]) extends Splitter(uid = uid) with
     val minLabelFract = getMinLabelFraction
     val maxLabels = getMaxLabelCategories
 
+    val labelCol = labelCounts.columns(0)
     val colCount = labelCounts.columns(1)
     val totalValues = labelCounts.agg(sum(colCount)).first().getLong(0).toDouble
     val labelsKeep = labelCounts
+      .sort(col(colCount).desc, col(labelCol))
       .filter(r => (r.getLong(1) / totalValues) >= minLabelFract)
-      .sort(col(colCount).desc)
       .take(maxLabels)
       .map(_.getDouble(0))
 
     val labelSet = labelsKeep.toSet
-    val labelsDropped = labelCounts.filter(r => !labelSet.contains(r.getDouble(0))).collect().map(_.getDouble(0)).toSet
+    val labelsDropped = labelCounts
+      .filter(r => !labelSet.contains(r.getDouble(0)))
+      .take(100)
```

Comment:
maybe instead of just keeping the first hundred dropped labels we should just keep a count of how many we dropped

Reply:
Good idea. We could also do both: number of dropped labels and the topK of them for some low K, so we can log it `dropped label1, label3, label10 ...`